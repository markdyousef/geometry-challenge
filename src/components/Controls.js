import React from "react";
import styled from "styled-components";
import Input from "./Input";

const ControlsContainer = styled.div`
  display: flex;
  justify-content: space-between;
  width: 340px;
  background: #fff;
  padding: 40px 30px;
  margin: 20px;
  border-radius: 2px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24);
  transition: all 0.3s cubic-bezier(0.25, 0.8, 0.25, 1);
  :hover {
    box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
  }
`;

export default class extends React.PureComponent {
  handleChange = event => {
    const { id, value } = event.target;
    // ensure non-negative number
    const length = value > -1 ? Number(value) : 0;
    const side = { [id]: length };
    this.props.changeSide(side);
  };
  render() {
    // [["a", 1], ["b", 2], ["c", 3]]
    const sides = Object.entries(this.props.sides);
    return (
      <ControlsContainer>
        {sides.map(side => (
          <Input
            id={side[0]}
            value={side[1]}
            key={`input_${side[0]}`}
            handleChange={this.handleChange}
          />
        ))}
      </ControlsContainer>
    );
  }
}
