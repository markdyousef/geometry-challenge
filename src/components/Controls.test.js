import React from "react";
import { shallow } from "enzyme";
import Controls from "./Controls";

it("renders Controls", () => {
  const wrapper = shallow(<Controls sides={{ a: 1, b: 1, c: 1 }} />);
  expect(wrapper).toMatchSnapshot();
});
