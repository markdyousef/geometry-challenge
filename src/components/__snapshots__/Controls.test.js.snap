// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Controls 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PureComponent
    sides={
        Object {
            "a": 1,
            "b": 1,
            "c": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <PureComponent
          handleChange={[Function]}
          id="a"
          value={1}
/>,
        <PureComponent
          handleChange={[Function]}
          id="b"
          value={1}
/>,
        <PureComponent
          handleChange={[Function]}
          id="c"
          value={1}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "input_a",
        "nodeType": "class",
        "props": Object {
          "handleChange": [Function],
          "id": "a",
          "value": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "input_b",
        "nodeType": "class",
        "props": Object {
          "handleChange": [Function],
          "id": "b",
          "value": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "input_c",
        "nodeType": "class",
        "props": Object {
          "handleChange": [Function],
          "id": "c",
          "value": 1,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <PureComponent
            handleChange={[Function]}
            id="a"
            value={1}
/>,
          <PureComponent
            handleChange={[Function]}
            id="b"
            value={1}
/>,
          <PureComponent
            handleChange={[Function]}
            id="c"
            value={1}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "input_a",
          "nodeType": "class",
          "props": Object {
            "handleChange": [Function],
            "id": "a",
            "value": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "input_b",
          "nodeType": "class",
          "props": Object {
            "handleChange": [Function],
            "id": "b",
            "value": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "input_c",
          "nodeType": "class",
          "props": Object {
            "handleChange": [Function],
            "id": "c",
            "value": 1,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
